db.py

```diff
@@ -32,20 +92,29 @@ def set_sqlite_pragma(dbapi_connection, connection_record):
     cursor.close()
 
 
-async def check_all_tables_exist(db_engine):
-    async with db_engine.begin() as conn:
-        for table in Base.metadata.tables.values():
-            result = await conn.execute(
-                text(f"SELECT name FROM sqlite_master WHERE type='table' AND name='{table.name}'"))
+async def check_all_tables_exist(session: Union[AsyncSession, Session]):
+    for table in Base.metadata.tables.values():
+        sql_query = f"SELECT name FROM sqlite_master WHERE type='table' AND name='{table.name}';"
+        if isinstance(session, AsyncSession):
+            result = await session.execute(text(sql_query))
+            if result.scalar() is None:
+                return False
+        else:
+            result = session.execute(text(sql_query))
             if result.scalar() is None:
                 return False
     return True
 
 
 async def create_db_and_tables():
-    async with engine.begin() as conn:
-        if await check_all_tables_exist(engine):
+    async with get_db_session() as session:
+        if await check_all_tables_exist(session):
             pass
         else:
-            await conn.run_sync(Base.metadata.drop_all)
-            await conn.run_sync(Base.metadata.create_all)
+            if isinstance(session, AsyncSession):
+                async with engine.begin() as conn:
+                    await conn.run_sync(Base.metadata.drop_all)
+                    await conn.run_sync(Base.metadata.create_all)
+            else:
+                Base.metadata.drop_all(bind=engine)
+                Base.metadata.create_all(bind=engine)
```

Comment:
Maybe this is more a bug fix that the existing files are currently dropped when a new schema is introduced

Reply:
@6R1M5L07H In production, the composition of database tables will never change. But during development it may help to quickly recreate the database in case of any changes in models.